test/Parse/ConditionalCompilation/language_version.swift

```diff
@@ -34,13 +48,21 @@
   %#^*&
 #endif
 
-#if swift(">=7.1") // expected-error {{unexpected platform condition argument: expected a unary comparison, such as '>=2.2'}}
+#if !swift(<1000.0)
```

Comment:
Please also add test cases for `#if compiler`, whose behavior is also changed by this PR.

Reply:
I noticed that tests for `#if compiler` were missing in the codebase, so I added tests for both operators in a new test file (https://github.com/apple/swift/pull/17960/commits/ccbc3f7dcc8bb41f3ab588b3167f1052875b4095)